### PR TITLE
Fix wrong edit incident link

### PR DESCRIPTION
**Pull request by cesarvarela**
Makes sure there is a default value for `incindetId` and `reportNumber`. 

Using Gatsby's dynamic routing might be more elegant but this would do for now.

### CODE REVIEW

File: site/gatsby-site/src/templates/cite.js [hunk 1]
```diff
@@ -181,7 +181,7 @@ function CitePage(props) {
                   <Button
                     variant="outline-primary"
                     className="me-2"
-                    href={'/incidents/edit?incident_id=' + incident.incident_id}
+                    href={'/incidents/edit?incidentId=' + incident.incident_id}
```
Comment (smcgregor) [inline site/gatsby-site/src/templates/cite.js:184]: Parameters should mirror their expression in the field definition. This should remain `incident_id` and this should be `const [incidentId] = useQueryParam('incident_id', withDefault(NumberParam, 1));` no?

Comment (cesarvarela) [reply site/gatsby-site/src/templates/cite.js:184]: Sure, it is my js mind that doesn't like snake_case, updating.

Comment (smcgregor) [reply site/gatsby-site/src/templates/cite.js:184]: https://github.com/responsible-ai-collaborative/aiid/issues/554

**Comment by cesarvarela** [comment]
Ready. Also updated the `report_number` field for consistency.

**Pull request merged**